cirrus/server/tests/test_experiment_recipes.py

```diff
@@ -4,11 +4,16 @@
 import requests
 
 from cirrus.experiment_recipes import RemoteSettings
+from cirrus.sdk import SDK
+from cirrus.settings import context
 
 
 class TestFetchRecipes(unittest.TestCase):
```

Comment:
Why unittest and not pytest?

Reply:
Usually it's prudent to avoid adding any additional dependencies until they're necessary.  Python has a robust builtin test framework that is used broadly across Nimbus, for instance by all of the Experimenter unit tests.  I wouldn't suggest incorporating pytest unless there's a specific need to.

Reply:
so currently unittest meets our testing requirements, and incorporating pytest would introduce an additional dependency without a specific need. However, we can keep pytest in mind for future considerations :)

Reply:
> Usually it's prudent to avoid adding any additional dependencies until they're necessary. Python has a robust builtin test framework that is used broadly across Nimbus, for instance by all of the Experimenter unit tests. I wouldn't suggest incorporating pytest unless there's a specific need to.

There's pytest being declared wthin some of these tests already. Also, a lot of the tests redefine the same variables for each test, pytest would solve this easily.  We have pytest for the Rust integration tests, now unit test here, and we will probably use pytest for the integration tests here. That seems like a lot of fragmentation for testing, when we could just use pytest everywhere and have a matching framework for all the tests.

> Sadly, the builtin methods in the Python unittest framework use camelcase.

No need for setup functions either with pytest.

Reply:
@jrbenny35 Yes super duper good points, we should absolutely be consistent in which patterns and frameworks we're using.  If we're already using some pytest constructs we should continue to, 100% agree 👍

Reply:
yeah, my mistake, I missed that, I think in starting tests I added, then started using unittest, this also requires some changes in other test files too- so added a ticket, will do follow up in here- https://github.com/mozilla/experimenter/issues/8993